src/main/java/com/ericsson/ei/subscription/SubscriptionRepeatDbHandler.java

```diff
@@ -82,70 +86,72 @@
-            LOGGER.debug(
-                    "New Subscription AggrId has not matched, inserting new SubscriptionId and AggrObjId to matched list.");
-            BasicDBObject document = new BasicDBObject();
-            document.put("subscriptionId", subscriptionId);
+            if (!updateExistingMatchedSubscriptionWithAggrObjId(subscriptionId, requirementId, aggrObjId)) {
+                LOGGER.error("Couldn't update SubscriptionMatched id.");
+                LOGGER.debug(
+                        "New Subscription AggrId has not matched, inserting new SubscriptionId and AggrObjId to matched list.");
+                BasicDBObject document = new BasicDBObject();
+                document.put("subscriptionId", subscriptionId);
 
-            ArrayList<String> aggrObjIdsList = new ArrayList<String>();
-            aggrObjIdsList.add(aggrObjId);
+                ArrayList<String> aggrObjIdsList = new ArrayList<String>();
+                aggrObjIdsList.add(aggrObjId);
 
-            BasicDBObject reqDocument = new BasicDBObject();
-            reqDocument.put(String.valueOf(requirementId), aggrObjIdsList);
+                BasicDBObject reqDocument = new BasicDBObject();
+                reqDocument.put(String.valueOf(requirementId), aggrObjIdsList);
 
-            document.put("requirements", reqDocument);
+                document.put("requirements", reqDocument);
 
-            LOGGER.debug("New Matched AggrIdObject update on Subscription to be inserted to Db: " + document);
-            boolean result = mongoDbHandler.insertDocument(dataBaseName, collectionName, document.toString());
-            if (result == false) {
-                throw new Exception("Failed to insert the document into database");
+                LOGGER.debug("New Matched AggrIdObject update on Subscription to be inserted to Db: " + document);
+                mongoDbHandler.insertDocument(dataBaseName, collectionName, document.toString());
             }
+        } catch (MongoWriteException e) {
+            LOGGER.error("Failed to insert the document into database");
+            e.printStackTrace();
         }
+
     }
 
     private boolean updateExistingMatchedSubscriptionWithAggrObjId(String subscriptionId, int requirementId,
-            String aggrObjId) throws Exception {
-        String subscriptionQuery = "{\"subscriptionId\" : \"" + subscriptionId + "\"}";
-        JsonNode updateDocJsonNode = mapper.readValue(
-                "{\"$push\" : { \"requirements." + requirementId + "\" : \"" + aggrObjId + "\"}}", JsonNode.class);
-        LOGGER.debug("SubscriptionId \"", subscriptionId +
-                "\" document will be updated with following requirement update object: "
-                        + updateDocJsonNode.toString());
-
-        JsonNode queryJsonNode = mapper.readValue(subscriptionQuery, JsonNode.class);
+            String aggrObjId) {
 
-        Document document = null;
         try {
```

Comment:
large try/catch is ugly code, why this? if try catches grow large they should be moved to it´s own function.

Reply:
I've extracted parts of this giant method to their own methods, handling the try catch separately